lib/Doctrine/DBAL/Query/BulkInsertQuery.php

```diff
@@ -0,0 +1,292 @@
+        $this->table = new Identifier($table);
+        $this->columns = $columns;
+    }
+
+    /**
+     * Adds a set of values to the bulk insert query to be inserted as a row into the specified table.
+     *
+     * @param array $values The set of values to be inserted as a row into the table.
+     *                      If no columns have been specified for insertion, this can be
+     *                      an arbitrary list of values to be inserted into the table.
+     *                      Otherwise the values' keys have to match either one of the
+     *                      specified column names or indexes.
+     * @param array $types  The types for the given values to bind to the query.
+     *                      If no columns have been specified for insertion, the types'
+     *                      keys will be matched against the given values' keys.
+     *                      Otherwise the types' keys will be matched against the
+     *                      specified column names and indexes.
+     *                      Non-matching keys will be discarded, missing keys will not
+     *                      be bound to a specific type.
+     *
+     * @throws \InvalidArgumentException if columns were specified for this query
+     *                                   and either no value for one of the specified
+     *                                   columns is given or multiple values are given
+     *                                   for a single column (named and indexed) or
+     *                                   multiple types are given for a single column
+     *                                   (named and indexed).
+     *
+     * @todo add support for expressions.
+     */
+    public function addValues(array $values, array $types = array())
+    {
+        if (empty($this->columns)) {
+            $this->addValuesInDbOrder($values, $types);
+        } else {
+            $this->addValuesByColumnDefinition($values, $types);
+        }
+    }
+
+    /**
+     * Executes this INSERT query using the bound parameters and their types.
+     *
+     * @return integer The number of affected rows.
+     *
+     * @throws \LogicException if this query contains more rows than acceptable
+     *                         for a single INSERT statement by the underlying platform.
+     */
+    public function execute(): int
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        $insertMaxRows = $platform->getInsertMaxRows();
+        $totalValues = count($this->values);
+
+        if ($insertMaxRows > 0 && $totalValues > $insertMaxRows && !$this->connection->isTransactionActive()) {
+            throw QueryException::transactionRequired();
+        }
+
+        $offset = 0;
+        $result = 0;
+        while ($offset < $totalValues) {
+            $values = array_slice($this->values, $offset, $insertMaxRows);
+            $parameters = array_slice($this->parameters, $offset, $insertMaxRows);
```

Comment:
Here should be something like:
$parameters = array_slice($this->parameters, $offset * count($this->values), $insertMaxRows * count($this->values));